src/devices/bus/cpc/magicsound.cpp

```diff
@@ -73,7 +73,9 @@ al_magicsound_device::al_magicsound_device(const machine_config &mconfig, const
 	m_dmac(*this,"dmac"),
 	m_timer1(*this,"timer1"),
 	m_timer2(*this,"timer2"),
-	m_current_channel(0), m_ramptr(nullptr), m_current_output(0)
+	m_ramptr(*this, "^^" RAM_TAG),
+	m_current_channel(0),
+	m_current_output(0)
```

Comment:
This is as bad as or worse than the current situation.  Devices shouldn’t make assumptions about the hosting system, particularly when it’s a slot device that could be installed in different types of systems with adapters.

Reply:
`"^^" RAM_TAG` also isn't going to work for passthrough interfaces.

The CPC slot has a priority line that allows expansion cards to conditionally override portions of the Z80 memory space. MAME didn't support that properly until late 2020, and serious work on CPC emulation ended about five years earlier, so `devices/bus/cpc` has a lot of evil banking hacks.